pallet-chainlink-feed/src/lib.rs

```diff
@@ -747,35 +748,35 @@ pub mod pallet {
 		/// - Removes the details for the previous round if it was superseded.
 		///
 		/// Limited to the oracles of a feed.
-		#[pallet::weight(
-            <SubmitWeight<T>>::new(
-                T::WeightInfo::submit_opening_round_answers().max(
-		            T::WeightInfo::submit_closing_answer(T::OracleCountLimit::get())
-                ),
-                oracle.clone(),
-                *feed_id,
-                *round_id,
-            ))]
+		#[pallet::weight((T::WeightInfo::submit_opening_round_answers().max(
+		    T::WeightInfo::submit_closing_answer(T::OracleCountLimit::get())
+		), DispatchClass::Operational))]
 		pub fn submit(
 			origin: OriginFor<T>,
-			// Specified oracle account
-			//
-			// TODO:
-			//
-			// This field is for calculating weights currently, remove this
-			// field after https://github.com/paritytech/substrate/issues/6316
-			oracle: T::AccountId,
 			#[pallet::compact] feed_id: T::FeedId,
 			#[pallet::compact] round_id: RoundId,
 			#[pallet::compact] submission: T::Value,
 		) -> DispatchResultWithPostInfo {
-			ensure_signed(origin)?;
+			let oracle = ensure_signed(origin)?;
 
 			with_transaction_result(|| -> DispatchResultWithPostInfo {
 				let mut feed = Feed::<T>::load_from(feed_id).ok_or(Error::<T>::FeedNotFound)?;
 				let mut oracle_status =
 					Self::oracle_status(feed_id, &oracle).ok_or(Error::<T>::NotOracle)?;
-				feed.ensure_valid_round(&oracle, round_id)?;
+
+				// Conditional PaysFee
+				let pays = match T::SubmitterPaysFee::get() {
+					SubmitterPaysFee::Always => Pays::Yes,
+					SubmitterPaysFee::FreeForValidRound => <Feed<T>>::load_from(feed_id)
+						.map(|feed| {
+							if feed.ensure_valid_round(&oracle, round_id).is_ok() {
+								Pays::No
+							} else {
+								Pays::Yes
+							}
+						})
+						.unwrap_or(Pays::Yes),
+				};
```

Comment:
This seems like overhead,
Can't we just check at the end? then it obviously work and we only need to convert `T::SubmitterPaysFee` into `PaysFee`?

Reply:
Yep, placed this block to the previous `ensure_valid_round` at LOL

Reply:
One thing that needs to consider is if the parameters of our checking will change in the previous logic, checking at the top is clean and at the end is also reasonable.